docs/reference/glossary.rst: New terms, reduce complexity.

jimmo opened the pull request.

jimmo committed:
docs/reference/glossary.rst: New terms, reduce complexity.

stinos commented on docs/reference/glossary.rst:
```diff
@@ -4,152 +4,190 @@ Glossary
+        extract data from them (or make a copy).
+
+    CircuitPython
+        An alternate version of MicroPython developed by `Adafruit Industries
```
I think `alternative` might be more correct here?

stinos commented on docs/reference/glossary.rst:
```diff
@@ -4,152 +4,190 @@ Glossary
+        However, a large subset of these modules require POSIX-like
+        environment (Linux, FreeBSD, MacOS, etc.; Windows may be partially
+        supported), and thus would work or make sense only on the `MicroPython
+        Unix port`. Some subset of modules are usable for the
```
Maybe add a short explanation that the main reason for this (i.e. most of it only being usable on POSIX-like) is that's it is implemented by ffi?

stinos commented on docs/reference/glossary.rst:
```diff
@@ -4,152 +4,190 @@ Glossary
+        :term:`CPython`'s ``python`` executable.
+
+    .mpy file
+        The output of the cross compiler. A compiled form of a :term:`.py` file
```
Also use `:term:` for 'cross compiler'?

jimmo commented:
> I think alternative might be more correct here?

You're absolutely right. However, I changed it to "A variant of MicroPython" which I think reads better.

Fixed the other comments too. Thanks!

mcauser commented:
Looks good. Found `acryonym` misspelled twice further down.
"acryonym" -> "acronym"

mcauser commented:
In upip: "A package manage" -> "A package manager"

jimmo commented:
> "acryonym" -> "acronym"
> In upip: "A package manage" -> "A package manager"

Fixed. Thanks!

mattytrentini commented:
These descriptions are clearer to me @jimmo, thanks!

dpgeorge commented:
Thanks a lot!  Merged in a93495b66d1e1101fffa3c8f2811d8750b5601f9 with some minor fixes to the cross references within the glossary itself.
